www/_template/reference/configuration.md

```diff
@@ -5,110 +5,325 @@ description: The Snowpack configuration API reference.
+Known dependencies to install with Snowpack.
 
-- Sets a `process.env.` environment variable inside the installed dependencies. If set to true (ex: `{NODE_ENV: true}` or `--env NODE_ENV`) this will inherit from your current shell environment variable. Otherwise, set to a string (ex: `{NODE_ENV: 'production'}` or `--env NODE_ENV=production`) to set the exact value manually.
+> Used for installing packages any dependencies that couldn't be detected by our automatic import scanner (ex: package CSS files).
 
-#### installOptions.treeshake `boolean`
+## config.extends
 
-- _Default:`false`, or `true` when run with `snowpack build`_
-- Treeshake your dependencies to optimize your installed files. Snowpack will scan your application to detect which exact imports are used from each package, and then will remove any unused imports from the final install via dead-code elimination (aka tree shaking).
+**Type**: `string`
 
-#### installOptions.installTypes `boolean`
+Inherit from a separate "base" config.
 
-- Install TypeScript type declarations with your packages. Requires changes to your [tsconfig.json](#typescript) to pick up these types.
+> Can be a relative file path, an npm package, or a file within an npm package. Your configuration will be merged on top of the extended base config.
 
-#### installOptions.namedExports `string[]`
+## config.exclude
 
-- _NOTE(v2.13.0): Snowpack now automatically supports named exports for most Common.js packages. This configuration remains for any package that Snowpack can't handle automatically. In most cases, this should no longer be needed._
-- Import CJS packages using named exports (Example: `import {useTable} from 'react-table'`).
-- Example: `"namedExports": ["react-table"]`
+**Type**: `string[]`
 
-#### installOptions.externalPackage `string[]`
+**Default**: `['**/node_modules/**/*', '**/web_modules/**/*', '**/.types/**/*']`
 
-- _NOTE: This is an advanced feature, and may not do what you want! Bare imports are not supported in any major browser, so an ignored import will usually fail when sent directly to the browser._
-- Mark some imports as external. Snowpack won't install them and will ignore them when resolving imports.
-- Example: `"externalPackage": ["fs"]`
+Exclude any files from the Snowpack pipeline.
 
-#### installOptions.packageLookupFields `string[]`
+Supports glob pattern matching.
 
-- Set custom lookup fields for dependency `package.json` file entrypoints, in addition to the defaults like "module", "main", etc. Useful for package ecosystems like Svelte where dependencies aren't shipped as traditional JavaScript.
-- Example: `"packageLookupFields": ["svelte"]`
+## config.knownEntrypoints
 
-#### installOptions.rollup `Object`
+**Type**: `string[]`
 
-- Snowpack uses Rollup internally to install your packages. This `rollup` config option gives you deeper control over the internal rollup configuration that we use.
+TODO: I think it's similar to plugin.knownEntrypoints
```

Comment:
@FredKSchott could use some pointers on what "config.knownEntrypoints" does

Reply:
not actually a user-facing configuration. Note that `SnowpackUserConfig` is the interface for what the user provides, while `SnowpackConfig` is the internal representation of config.